wallet/db_common.h

```diff
@@ -34,6 +34,10 @@ struct db {
 	/* Were there any modifying statements in the current transaction?
 	 * Used to bump the data_version in the DB.*/
 	bool dirty;
+
+	/* The current DB version we expect to update if changes are
+	 * committed. */
+	u32 data_version;
```

Comment:
u64 not u32?

Reply:
Yes, `intval` in the `vars` table is defined as an `INTEGER` which gets translated into a `u32` in postgres and the size has to match in non-sqlite3 databases. So I thought I shouldn't be too picky about this value.

Alternatives are 1) creation of a new table, or 2) adding a new field `bigintval` to the `vars` table. If this is a problem I'd prefer the second. But really all we're guaranteeing is that there is enough variety to identify if we skipped writes or have lost some. In particular we can't do much if we're off by more than 2: if the plugin missed an update we must take a snapshot, if the DB moved back to an earlier point in time we must abort. We just need to have the probability of accidentally using the same version again small enough (1/2**32 seems good enough for me 😉).

Reply:
sweet, ok. that's a good thing to know about int -> u32 in postgres.